### PR TITLE
PR: tp1

### PR DESCRIPTION
**Pull request by elyskimo**
This is a test PR

### CODE REVIEW

File: ts/apps/tournament/src/app/api/tournament-api.ts [hunk 1]
```diff
@@ -6,8 +6,22 @@ import { v4 as uuidv4 } from 'uuid';
 const tournamentRepository = new TournamentRepository();
 
 export const postTournament = (req: Request, res: Response) => {
+
+  if (!Object.keys(req.body).length) {
+     res.status(400);
+     return res.send({message: "can't create tournament, the name has been given"});
```
Comment (zygof) [inline ts/apps/tournament/src/app/api/tournament-api.ts]: fix message : "can't create tournament, the name is missing"

File: ts/apps/tournament/src/app/api/tournament-api.ts [hunk 2]
```diff
@@ -6,8 +6,22 @@ import { v4 as uuidv4 } from 'uuid';
...
+  }
+  if (!req.body.name.length) {
```
Comment (zygof) [inline ts/apps/tournament/src/app/api/tournament-api.ts]: add condition : 
issue 1 : "name?.length"
issue 2 "req.body.name !== null"

File: ts/apps/tournament/src/app/repository/tournament-repository.ts [hunk 3]
```diff
@@ -10,4 +10,8 @@ export class TournamentRepository {
   public getTournament(tournamentId: string): Tournament {
     return this.tournaments.get(tournamentId);
   }
+
+  public getTournamentByName(tournamentName: string): Tournament {
+     return Array.from(this.tournaments.values()).filter((item: Tournament) => item.name === tournamentName)[0];
```
Comment (zygof) [inline ts/apps/tournament/src/app/repository/tournament-repository.ts]: suggestion : Array.find()
return the first element matching.

File: ts/apps/tournament/src/app/api/tournament-api.ts [hunk 4]
```diff
@@ -33,7 +33,12 @@ export const getTournament = (req: Request, res: Response) => {
   const id = req.params['id'];
 
   const tournament = tournamentRepository.getTournament(id);
-
+if (tournament) {
   res.status(200);
   res.send(tournament);
+}else{
+  res.status(400);
```
Comment (zygof) [inline ts/apps/tournament/src/app/api/tournament-api.ts]: fix : error 404

File: ts/apps/tournament/src/testing/tournament.spec.ts [hunk 5]
```diff
@@ -50,4 +51,10 @@ describe('/tournament endpoint', () => {
       expect(body.message).toEqual("can't create tournament, the name already exist");
     });
   });
+  describe('[GET] when user want tournament descriptions', () => {
+    it('tournament does not exist', async () => {
+      const { body } = await request(app).get('/api/tournaments/' + exampleTournament.id ).expect(400);
```
Comment (zygof) [inline ts/apps/tournament/src/testing/tournament.spec.ts]: fix :
- error 404

feat :
- add test if id null
- add test if id is empty
